swing/src/net/sf/openrocket/gui/dialogs/ComponentAnalysisDialog.java

```diff
@@ -134,7 +130,9 @@ public ComponentAnalysisDialog(final RocketPanel rocketPanel) {
 
 		//// Wind direction:
 		panel.add(new JLabel(trans.get("componentanalysisdlg.lbl.winddir")), "width 120lp!");
-		panel.add(new UnitSelector(theta, true), "width 50lp!");
+		EditableSpinner spinner = new EditableSpinner(theta.getSpinnerModel());
```

Comment:
The spinners appear very large in the UI. I just pushed a change to the unstable branch where I have updated the MigLayout of `panel` to set the column widths in the layout ([see commit](https://github.com/openrocket/openrocket/commit/e62cd4c6db8571e6db9abf66cf818e1b306a4977)). This ensures that you don't have to set the widths explicitly (notice that the `120lp` width constraint from the labels is now defined in the layout instead of repeating it for every label).

So, could you first merge the changes from the latest unstable (sync your GitHub repo with the openrocket unstable branch, `git checkout unstable && git fetch && git pull` your local unstable branch, and merge those changes in your `issue-2258` branch (`git checkout issue-2258 && git merge unstable`). You can also do this all easily in IntelliJ, but it's difficult to explain here...

Then, change the panel initialization to set appropriate column widths:
```Java
JPanel panel = new JPanel(new MigLayout("fill", "[120lp][70lp][50lp][]"));
```

Here I set the spinner to 70lp width and unit selector to 50lp. You can play around with it to see what's best.